bridgetown-core/lib/bridgetown-core/commands/concerns/actions.rb

```diff
@@ -77,23 +77,34 @@ def apply_from_url(url)
 
       private
 
+      def remote_file(arg)
+        if arg.end_with?(".rb")
+          arg.split("/").yield_self do |segments|
+            arg.sub!(%r!/#{segments.last}$!, "")
+            segments.last
+          end
+        else
+          "bridgetown.automation.rb"
+        end
+      end
+
       # TODO: option to download and confirm remote automation?
       def transform_automation_url(arg)
         return arg unless arg.start_with?("http")
 
-        remote_file = if arg.end_with?(".rb")
-                        arg.split("/").yield_self do |segments|
-                          arg.sub!(%r!/#{segments.last}$!, "")
-                          segments.last
-                        end
-                      else
-                        "bridgetown.automation.rb"
-                      end
+        remote_file = remote_file(arg)
+
+        tree_regex = %r!https://github\.com/(?<path>.*/.*)/tree/(?<branch>.*)/?!
+        match = tree_regex.match(arg)
 
         if arg.start_with?("https://gist.github.com")
           return arg.sub(
             "https://gist.github.com", "https://gist.githubusercontent.com"
           ) + "/raw/#{remote_file}"
+        elsif match
+          return arg.sub(
+            tree_regex, "https://raw.githubusercontent.com"
+          ) + "/#{match[:path]}/#{match[:branch]}/#{remote_file}"
         elsif arg.start_with?("https://github.com")
```

Comment:
Do we need still need to check GitHub with `start_with?` if we're doing the match above?

Reply:
The match above wont match with a regular master branch because it doesnt include `tree`.

However, we could make a base regex for github:

IE:

```ruby
github_regex = %r!https://github\.com!
```

And then use Regex concatenation to drop it into the tree regex like so:

```ruby
github_tree_regex = %r!#{github_regex}/(?<path>.*/.*)/tree/(?<branch>.*)/?!
```

First we want to run the matches up front before we mess with the URL.
Then we will always replace Github on github_regex match
Something like the following:

```ruby
github_match = github_regex.match(arg)
github_tree_match = github_tree_regex.match(arg)

elsif github_match
  new_url = arg.sub(github_regex, "https://raw.githubusercontent.com")
  
  return new_url + "/master/#{remote_file}" if github_tree_match.nil?

  return new_url + "/#{match[:path]}/#{match[:branch]}/#{remote_file}"
end
```

Reply:
On further review, we actually dont need the `match[:path]` we could simple do `#{match[:branch}/#{remote_file}`

Reply:
That sounds like a good approach to me. Can you go ahead and update your branch with the conditional logic consolidated like that? 👍

Reply:
Absolutely, working on it now, the issue I'm running into is I have to sub "/tree/" but I dont want to run the risk of having the user have a directory named `/tree/` so it may be slightly more complicated than expected. Ill add a test for making sure you dont conflic with a `/tree/` directory.